src/acq/adc.c

```diff
@@ -483,21 +489,16 @@ enum bl_error bl_acq_adc_configure(bl_acq_adc_t *adc,
 		return BL_ERROR_HARDWARE_CONFLICT;
 	}
 #else
-	if (oversample > 0) {
-		uint32_t cfgr2 = ADC_CFGR2(adc->base);
-		cfgr2 &= ~(ADC_CFGR2_JOVSE | ADC_CFGR2_OVSR_MASK |
-				ADC_CFGR2_OVSS_MASK | ADC_CFGR2_TROVS |
-				ADC_CFGR2_ROVSM);
+	if ((oversample == 0) && (shift > 0)) {
+		return BL_ERROR_HARDWARE_CONFLICT;
+	}
 
-		cfgr2 |= ADC_CFGR2_OVSR_VAL(oversample - 1);
-		cfgr2 |= ADC_CFGR2_OVSS_VAL(shift);
-		cfgr2 |= ADC_CFGR2_ROVSE;
+	if (oversample > 8) {
+		return BL_ERROR_OUT_OF_RANGE;
+	}
 
-		ADC_CFGR2(adc->base) = cfgr2;
-	} else if (shift > 0) {
-		return BL_ERROR_HARDWARE_CONFLICT;
-	} else {
-		ADC_CFGR2(adc->base) &= ~ADC_CFGR2_ROVSE;
+	if (shift > 8) {
+		return BL_ERROR_OUT_OF_RANGE;
```

Comment:
a bit picky, but if return the same out of range error, why not combine them together?

Reply:
We could separate them in the future, so I treat them as different errors, but I didn't think they were important enough to have their own error codes.